flutter/lib/src/default_integrations.dart

```diff
@@ -494,3 +497,66 @@ class LoadReleaseIntegration extends Integration<SentryFlutterOptions> {
         .replaceAll('\u{0000}', '');
   }
 }
+
+/// Enrich [SentryTransaction] objects with native app data for mobile vitals.
+class MobileVitalsIntegration extends Integration<SentryFlutterOptions> {
+  MobileVitalsIntegration(this._nativeWrapper);
+
+  final SentryNativeWrapper _nativeWrapper;
+
+  @override
+  FutureOr<void> call(Hub hub, SentryFlutterOptions options) {
+    if (options.autoAppStartFinish) {
+      SchedulerBinding.instance?.addPostFrameCallback((timeStamp) {
+        options.appStartFinish = DateTime.now();
+      });
+    }
+
+    options.addEventProcessor(
+        _NativeAppStartEventProcessor(_nativeWrapper, options));
+
+    options.sdk.addIntegration('mobileVitalsIntegration');
+  }
+}
+
+class _NativeAppStartEventProcessor extends EventProcessor {
+  _NativeAppStartEventProcessor(this._nativeWrapper, this._options);
+
+  final SentryNativeWrapper _nativeWrapper;
+  final SentryFlutterOptions _options;
+
+  var _didFetchAppStart = false;
+
+  @override
+  FutureOr<SentryEvent?> apply(SentryEvent event, {hint}) async {
+    final appStartFinishTime = _options.appStartFinish;
+
+    if (appStartFinishTime != null &&
+        event is SentryTransaction &&
+        !_didFetchAppStart) {
+      _didFetchAppStart = true;
+
+      final nativeAppStart = await _nativeWrapper.fetchNativeAppStart();
```

Comment:
Maybe this should go with a try/finally

Reply:
This method does not throw, as the possible error sources (json/native bridge) are handled internally and logged. Or do you mean something else?

Reply:
Makes sense, `fetchNativeAppStart` already does a try catch internally.